### PR TITLE
Unintended Feature Fix: Stops Xeno Fruit's of the same type stacking.

### PR DESCRIPTION
**Pull request by Moonshanks**
# About the pull request

Adds a "can_consume_check" which checks if you are under the effects of a fruit before applying the effect stopping fruits of the same type stacking.

# Explain why it's good for the game

Stops mach 12 drones.

Also provides a better way to check for stacked fruit effects which will be useful into the future.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

https://youtu.be/Yge37UMPYAU

</details>

# Changelog

:cl:
fix: The effects from the alacrit fruit no longer stack.
/:cl:

### CODE REVIEW

**Comment by Moonshanks** [comment]
@BeagleGaming1 is the most patient saint known to man. His service should be rewarded with a good job sticker.

File: code/modules/cm_aliens/structures/fruit.dm [hunk 1]
```diff
@@ -28,6 +28,8 @@
 	var/mob/living/carbon/xenomorph/bound_xeno // Drone linked to this fruit
 	var/fruit_type = /obj/item/reagent_container/food/snacks/resin_fruit
 
+
```
[the remark this answers is not in the document]
Comment (Moonshanks) [reply code/modules/cm_aliens/structures/fruit.dm:31]: Your mother.

File: code/modules/cm_aliens/structures/fruit.dm [hunk 2]
```diff
@@ -285,13 +285,10 @@
 	var/speed_buff_amount = 0.4
 	var/speed_duration = 15 SECONDS
 
-/obj/effect/alien/resin/fruit/speed/prevent_consume(mob/living/carbon/xenomorph/xeno)
+/obj/effect/alien/resin/fruit/speed/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)
 	if(LAZYISIN(xeno.modifier_sources, XENO_FRUIT_SPEED))
 		to_chat(xeno, SPAN_XENOWARNING("You're already under the effects of this fruit, go out and kill!"))
```
[the remark this answers is not in the document]
Comment (Moonshanks) [reply code/modules/cm_aliens/structures/fruit.dm]: done... but I think I made one extra recipient. Let me go ahead and fix that.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
As we discussed on discord: change it to stop all stacking of individual types of fruits please.

**Comment by Moonshanks** [comment]
> As we discussed on discord: change it to stop all stacking of individual types of fruits please.

Yes boss

**Comment by Moonshanks** [comment]
Fixed, tested, working on all fruits.

File: code/modules/cm_aliens/structures/fruit.dm [hunk 3]
```diff
@@ -125,7 +125,13 @@
...
 	update_icon()
 
-/obj/effect/alien/resin/fruit/proc/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)
+/obj/effect/alien/resin/fruit/proc/can_consume_check(mob/living/carbon/xenomorph/recipient)
+	if(LAZYISIN(recipient.modifier_sources, name))
+		to_chat(recipient, SPAN_XENOWARNING("You're already under the effects of this fruit, go out and kill!"))
+		return
+
+/obj/effect/alien/resin/fruit/proc/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)//why do we bother making this if we're going to overite it later?
+	can_consume_check(recipient)
```
Comment (Watermelon914) [inline code/modules/cm_aliens/structures/fruit.dm:134]: This isn't exactly going to do anything, you need to return a boolean value and do an if check.
Then early return if the check does not pass.

Also, the name of your function is a bit verbose, `can_consume` is a good enough name.

File: code/modules/cm_aliens/structures/fruit.dm [hunk 4]
```diff
@@ -189,6 +195,7 @@
 
 
 /obj/effect/alien/resin/fruit/greater/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)
+	can_consume_check(recipient)
```
Comment (Watermelon914) [inline code/modules/cm_aliens/structures/fruit.dm:198]: Need to do this to the rest of them as well.

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**